### PR TITLE
Projectiles feedback on battle

### LINKED ISSUE #354: Show attack feedback for battle
A feedback to communicate the user that an attack is being performed, showing caster and target of said attack, is needed.

### PR DESCRIPTION
**Pull request by tkz00**
Closes #354 

⚠️ Needs these backend PRs to be merged:
- [Fix history occasionally removing entries](https://github.com/lambdaclass/mirra_backend/pull/481)
- [Fix dead units targeting in delayed effects](https://github.com/lambdaclass/mirra_backend/pull/477)

## Motivation

Feedback that communicates the user the start of an attack (the EFFECT_TRIGGER) and the hit (EFFECT_HIT) or miss (EFFECT_MISS) is needed to have a better understanding of the battle.

## Summary of changes

- Added projectiles pooler reference to battle manager.
- On the PlayOutSteps method:
    - Iterate through all the EFFECT_TRIGGER actions, and call the projectiles pooler TriggerProjectile method to spawn an arrow pointing from the caster unit to each of the targets, with a green color if they are the same team and red if they are enemies, this arrow has some transparency.
    - Call the projectiles pooler ProjectileHit method whenever an EFFECT_HIT or EFFECT_MISS action is read from the battle history, which removes the transparency from the attack feedback arrows, convening the attack is taking place, .2f seconds after, the arrow is made disabled and up for use for the pooler.

## How has this been tested?

Make sure both the backend PRs listed on the top of the description are merged to your server.

Play out a battle, and check attacks, buffs, heals and any action a unit takes is first "targeted" with the transparent arrow and then becomes fully opaque when it hit's or misses, take the bottom video for reference.

Be aware that the time between the transparent arrow appearing and it becoming opaque is defined in the skills.json of the backend, the field of this delay is ```initial_delay``` inside each ```effect``` of the skill, if this value is 0 then the arrow will spawn with full opaqueness.

https://github.com/lambdaclass/champions_of_mirra/assets/67983386/a6051e34-5fe1-4c69-9a4b-36d66e8737ba

### CODE REVIEW

**Review by martinacantaro — APPROVED** [review]
Functional review approved.
After I set an initial_delay higher than 0 in the skills.json config, I was able to see the transparency between targeting and on_hit

**Review by martinacantaro — APPROVED** [review]
Approved again after the conflict fix

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 1]
```diff
@@ -88,7 +93,80 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
...
+						targetTeam = 2;
+					}
+
+					if(casterTeam == targetTeam) {
+						projectileColor = Color.green;
+					}
+
+					projectilesPooler.ProjectileHit(casterUnit, targetUnit, projectileColor);
+				}
+			}
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:169]: This duplicate code could be abstracted into another method

**Review by ncontinanza — CHANGES_REQUESTED** [review]

File: client/Assets/Scripts/Battle/BattleManager.cs (hunk 1 above)
Comment (tkz00) [reply client/Assets/Scripts/Battle/BattleManager.cs]: Yess! You are right, this has been fixed, and the code reordered a bit to improve clarity and redeability.

**Review by ncontinanza — APPROVED** [review]

**Review by martinacantaro — APPROVED** [review]
functional review approved!

**Pull request merged**